### DISCUSSION TITLE
[SOLVED] weird behavior: Tasmota on ESP8266 + Hichi IR + Darkness = no data.

### DISCUSSION THREAD

**Discussion by kyle-mge** [arendst/Tasmota#21089] [Q&A] [answered]
Hi guys,

I'm actually more than confused where the issue is coming from and hope you guys have the information to lead me in the correct direction.
I'm running Tasmota (flashed today on 13.4.0 to eliminate possible errors) on an ESP8266. Connected the Hichi IR ttl-Interface using it as a Smart-Meter-Interface to read my Logarex LK13BE803039 (Script as used in the Docs). Weird: it was running for approx. two years without any issues. It is set up like explained here: https://tasmota.github.io/docs/Smart-Meter-Interface/#logarex-lk13be803039

But the weird thing though is: my Meter is installed in an extra cabinet outside in front of the house. If I leave the door open of the cabinet and there's the normal surrounding ligth, I receive data, steady, stable, correct. As soon as I close the door and it's dark inside the cabinet, I don't receive any data anymore. As soon, as I open the door again, I see updated data.
The door is not blocking anything in terms of cabling, the only change is, that it is just dark inside.

I always thought, as the LEDs of the Logarex-Meters are not that strong, that it should be way better when dark.

I know it sounds weird and to be honest, I'm quite confused where to start to get this resolved.
Many thanks for your support in advance and cheers
Matze.

**Comment by sfromis** [comment]
On an operational level, what do you mean by "I don't receive any data anymore" - seen where?

What I'm wondering if the cabinet door could affect Wifi signal propagation (including reflections), especially if there are metal parts, which very often is the case around panels for electricity. In that case, physical placement/orientation of the antenna could be factor; even small differences can have an impact.

**Reply by kyle-mge** [reply]
@sfromis 

Thanks for your quick answer!
With receiving no data anymore I meant, that's there's no comm und the serial interface anymore. 
In the webinterface, I can see the (Live-) Updates from the data sent by the meter, especially the Live-Consumption (being updated roughly every second). As soon as I close the door, the data being shown freezes.
![image](https://github.com/arendst/Tasmota/assets/80550116/99b17b9f-6573-4ec8-b2f6-2f2bb2d53fd4)


WiFi coverage is no issue, never had the issue the past couple of years and with the door closed, I still can access the ESP through Webinterface and MQTT without issues.
![image](https://github.com/arendst/Tasmota/assets/80550116/e7a44cea-3178-48b3-adb2-37d7e29da43b)

**Reply by sfromis** [reply]
As Tasmota cannot react to where the cabinet door is, the difference has to be in the physical world around the hardware. It could still be related to combinations of radio signals and the cable to the IR head acting as an antenna picking up signals differently, and maybe more interference with the door closed.

Or maybe you are pushing/pulling the IR head when you close the door. Another speculation could be IR sources and reflections acting differently.

**Reply by kyle-mge** [reply]
Thanks @sfromis

that validates my thoughts, even though I'm still confused why this is happening....
I will pick this up with Hichi as the developer of the IR head... Hopefully to find further ideas to get this resolved.


Thanks for your suggestions so far!!!

**Comment by Noschvie** [comment]
> on 13.4.0 

Which Tasmota version did you use before?

**Reply by kyle-mge** [reply]
Used 9.5.0, 12.5.0 and now 13.4.0.

Funny thing, I just stumbled over my 9.5.0 I had running, flashed it and.... same result... So I assume, it's not a Tasmota-issue, because 9.5.0 and quite some time the 12.5.0 worked....

**Comment by kyle-mge** [comment, marked as answer]
Ok, so this was a quick one to get it resolved....
I did not expect that the LED will "age" or gets worn out... I changed the IR head with a new one and the issues are completely gone. So in case someone might will face the same issue, the head did not just stop working but started to send wrong data until the conditions needed to be perfect just to receive anything (perfectly aligned + perfect light conditions). With the new IR head, there was not even the need to perfectly align but just to place it on the IR interface.

Thanks for your suggestions guys.
Cheers, Matze.
